core/src/main/java/com/uber/m3/tally/ScopeImpl.java

```diff
@@ -42,14 +43,14 @@
     private ScheduledExecutorService scheduler;
     private Registry registry;
 
-    // ConcurrentHashMap nearly always allowing read operations seems like a good
-    // performance upside to the consequence of reporting a newly-made metric in
-    // the middle of looping and reporting through all metrics. Therefore, we only
-    // synchronize on these maps when having to allocate new metrics.
-    private final Map<String, CounterImpl> counters = new ConcurrentHashMap<>();
-    private final Map<String, GaugeImpl> gauges = new ConcurrentHashMap<>();
-    private final Map<String, TimerImpl> timers = new ConcurrentHashMap<>();
-    private final Map<String, HistogramImpl> histograms = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, CounterImpl> counters = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, GaugeImpl> gauges = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, HistogramImpl> histograms = new ConcurrentHashMap<>();
+
+    // TODO validate if needs to be concurrent at all
+    private final ConcurrentLinkedQueue<MetricBase> reportingQueue = new ConcurrentLinkedQueue<>();
```

Comment:
Can remove the TODO, it needs the concurrency because metrics can be both created concurrently, and created and reported upon concurrently

Reply:
Cleaned up in #74